packages/app/src/app/utils/blocker.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * This is a waiting promise that only resolves when the given value is done
+ * initializing. It waits.
+ */
+export function blocker<T>(): {
```

Comment:
@CompuIves Extract `Props` and make this an arrow function

```suggestion
type Props<T> = {
  promise: Promise<T>;
  resolve: (value: T) => void;
}

export const blocker<T> = () => {
```